docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,213 @@
+
+	    DotNetCountLabel.Text = $"Number of .NETs on dotnetfoundation.com: {count}";
+
+	    NetworkProgressBar.IsEnabled = false;
+	    NetworkProgressBar.Visbility = Visibility.Collapsed;
+	}
+
+Web service snippet (ASP.NET MVC):
+
+.. code-block:: c#
+
+	private readonly HttpClient _httpClient = new HttpClient();
+
+	[HttpGet]
+	[Route("DotNetCount")]
+	public async Task<int> GetDotNetCountAsync()
+	{
+	    // Suspends GetDotNetCountAsync() to allow the caller (the web server) to accept another request,
+	    // rather than blocking on this one.
+	    var html = await _httpClient.DownloadStringAsync("https://dotnetfoundation.org");
+
+	    return Regex.Matches(html, ".NET").Count;
+	}
+
+Example (VB)
+------------
+
+These are the VB-equivalent code snippets from above.
+
+Client app snippet (Universal Windows App):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	Private Async Sub SeeTheDotNets_Click(sender As Object, e As RoutedEventArgs)
+
+	    ' Capture the task handle here so we can await it later.
+	    Dim getDotNetFoundationHtmlTask As Task(Of String) = _httpClient.GetStringAsync("http://dotnetfoundation.org")
+
+	    ' Any other work on the UI thread can be done here, such as enabling a Progress Bar.
+	    NetworkProgressBar.IsEnabled = true
+	    NetworkProgressBar.Visibility = Visibility.Visible
+
+	    ' The await operator suspends SeeTheDotNets_Click, returning control to its caller.
+	    ' This is what allows the app to be responsive and not hang on the UI thread.
+	    Dim html As String = Await getDotNetFoundationHtmlTask
+	    Dim count As Integer = Regex.Matches(html, ".NET").Count
+
+	    DotNetCountLabel.Text = $"Number of .NETs on dotnetfoundation.com: {count}"
+
+	    NetworkProgressBar.IsEnabled = false
+	    NetworkProgressBar.Visbility = Visibility.Collapsed
+	End Sub
+
+Web Service snippet (ASP.NET MVC):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	<HttpGet>
+	<Route("MicrosoftCount")>
+	Public Async Function GetDotNetCountAsync() As Task(Of String)
+
+	    ' Suspends GetDotNetCountAsync() to allow the caller (the web server) to accept another request,
+	    ' rather than blocking on this one.
+	    Dim html As String = Await _httpClient.GetStringAsync("https://dotnetfoundation.org");
+
+	    Return Regex.Matches(html, ".NET").Count
+	End Function
+
+More on Task and Task<T>
+------------------------
+
+As mentioned before, Tasks are constructs used to represent operations which you will need to wait on to complete.
+
+* ``Task`` represents a single operation which does not return a value.
+* ``Task<T>`` represents a single operation which returns a value of type ``T``.
+
+It's important to reason about Tasks of abstractions of work happening in the background, and *not* dedicated threads on the thread pool, because in the context of I/O-bound code, they aren't new threads at all!
```

Comment:
> Tasks of abstractions of work

That's a bit vague.

"they" is a bit vague. Tasks are not thread ever. I think you want to focus on "using" threads, not "being" threads.